spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonApplicationContextInitializer.java

```diff
@@ -0,0 +1,38 @@
+package org.springframework.cloud.netflix.ribbon;
+
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+
+import java.util.List;
+
+/**
+ * Responsible for eagerly creating the child application context holding the Ribbon
+ * related configuration
+ *
+ * @author Biju Kunjummen
+ */
+public class RibbonApplicationContextInitializer
```

Comment:
Yes, makes sense. This class does that though - taking in a list of Ribbon Client names and initializing the application contexts - would a different name help?